改进日志 1. 允许强行等待日志写完 2. 线程安全 3. 允许设置日志最大大小

walterlv opened the pull request:
1. 允许强行等待日志写完
2. 线程安全
3. 允许设置日志最大大小

walterlv committed:
允许多个进程同时写一个日志文件

walterlv committed:
允许限制日志文件的大小

walterlv committed:
为日志添加关闭时的线程安全

lindexi commented on src/Utils/Walterlv.Logger/Core/AsyncOutputLogger.cs:
```diff
@@ -128,5 +157,18 @@ private async void StartLogging()
+        /// <returns>可等待对象。</returns>
+        protected async Task WaitFlushingAsync()
+        {
+            _waitForEmptyTaskCompletionSource = new TaskCompletionSource<object?>();
```
这不是线程安全

lindexi commented on src/Utils/Walterlv.Logger/Core/AsyncOutputLogger.AsyncQueue.cs:
```diff
@@ -18,6 +18,8 @@ public AsyncQueue()
                 _queue = new ConcurrentQueue<T>();
             }
 
+            public int Count => _queue.Count;
```
这里推荐放在 https://github.com/dotnet-campus/AsyncWorkerCollection/blob/master/AsyncWorkerCollection/AsyncQueue.cs

walterlv replied:
见这里：https://github.com/dotnet-campus/AsyncWorkerCollection/pull/7

walterlv committed:
确保 AsyncOutputLogger 的等待线程安全

walterlv committed:
加注释

walterlv committed:
调整时间，因为单元测试不过
https://github.com/walterlv/Walterlv.Packages/pull/10/checks?check_run_id=821753049

walterlv committed:
修复单元测试
